scrapy/pipelines/files.py

```diff
@@ -460,4 +461,11 @@ def item_completed(self, results, item, info):
     def file_path(self, request, response=None, info=None):
         media_guid = hashlib.sha1(to_bytes(request.url)).hexdigest()
         media_ext = os.path.splitext(request.url)[1]
+        # Handles empty and wild extensions by trying to guess the
+        # mime type then extension or default to empty string otherwise
+        if media_ext not in mimetypes.types_map.keys():
+            media_ext = ''
+            media_type = mimetypes.guess_type(request.url)[0]
+            if media_type:
+                media_ext = mimetypes.guess_extension(media_type)
```

Comment:
The [coverage report](https://codecov.io/gh/scrapy/scrapy/pull/3954) seems to suggest that everything is covered, however I think the test you introduced does not reach this inner `if` statement. If that is so, could you add one more test that does enter the inner `if` statement?

Reply:
Yeah it actually didn't reach it.. Added a test to cover this code